### PR TITLE
constify isNested and isOverloadable

### PR DESCRIPTION
**Pull request by benjones**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by kinke** [comment]
Thx for keeping the C++ headers in sync; that has been forgotten in previous constifications. 👍

**Comment by Basile-z** [comment]
@benjones, merging was apparently blocked by one of the project tested by Buildkite. 
It has been deactivated yesterday so the PR should be all green after amending (or even rebase). After what you can request the merge label again.

**Comment by Basile-z** [comment]
Or a dmd team member can do that to if the option allowing them to edit has not been deactivated.

**Pull request merged**
